homeassistant/components/weatherflow_cloud/config_flow.py

```diff
@@ -49,11 +49,15 @@ async def async_step_reauth_confirm(
             errors = await _validate_api_token(api_token)
             if not errors:
                 # Update the existing entry and abort
-                return self.async_update_reload_and_abort(
-                    self._get_reauth_entry(),
-                    data={CONF_API_TOKEN: api_token},
-                    reload_even_if_entry_is_unchanged=False,
-                )
+                if existing_entry := self.hass.config_entries.async_get_entry(
+                    self.context["entry_id"]
+                ):
+                    return self.async_update_reload_and_abort(
+                        existing_entry,
+                        data={CONF_API_TOKEN: api_token},
+                        reason="reauth_successful",
+                        reload_even_if_entry_is_unchanged=False,
+                    )
```

Comment:
why this change?

Reply:
I'm pretty sure I saw it in another integration. Is it wrong?

Reply:
It also appears the file is 100% covered by tests so I guess Its at least being tested. Hard cause I did this so long ago to remember things in my advanced age. :)